lib/cache/resource.js

```diff
@@ -68,7 +68,12 @@ function getNestedResourcesFromCache(obj) {
             continue;
         }
         if (obj[key].hasOwnProperty('key') && obj[key].hasOwnProperty('value') && obj[key].hasOwnProperty('expiration')) {
-            obj[key] = getFromCache.call(this, obj[key].key);
+            var cacheValue = getFromCache.call(this, obj[key].key);
+
+            //Only set the object value if getFromCache does not return an undefined object
+            if (cacheValue) {
```

Comment:
Technically, falsy values should be cacheable. So we should change this to be a check for `undefined`:

    if (typeof cacheValue !== 'undefined') {